synapse/metrics/metric.py

```diff
@@ -105,10 +156,12 @@ def render(self):
         value = self.callback()
 
         if self.is_scalar():
-            return ["%s %.12g" % (self.name, value)]
+            return list(self._render_for_labels([], value))
 
-        return ["%s%s %.12g" % (self.name, self._render_key(k), value[k])
-                for k in sorted(value.keys())]
+        return flatten(
+            self._render_for_labels(k, value[k])
+            for k in sorted(value.keys())
+        )
```

Comment:
Is `k` here really going to be a list? Why do we need to flatten this when we didn't previously?

Reply:
Oh, I'm an idiot, nevermind

Reply:
:)